### ISSUE TITLE
Should we force eval3D when running with UT?

### ISSUE THREAD

**Issue by atonderski** [nerfstudio-project/gsplat#711]
Hi,

Currently it is possible to run with UT but not eval3D. It will not raise any errors but will also not propagate any gradients to the "2D" rasterizer causing broken trainings. I assume there was a thought behind separating the flags, perhaps to enable 3d eval without UT projection?

If I understand the matter correctly the simplest fix is perhaps an "if withUT: assert eval3D"?

**Comment by liruilong940607** [comment]
For the forward pass (only rendering), UT can work with both eval2d / eval3d. It's just when we want to compute gradient during training, we have to use eval3d for UT. 

I think your proposal makes sense. we could add a if-else assert in the training script to enforce that.
